app/src/main/java/com/kamwithk/ankiconnectandroid/ankidroid_api/IntegratedAPI.java

```diff
@@ -78,8 +80,22 @@ public Long addNote(final Map<String, String> data, String deck_name, String mod
         }
     }
 
-    public String storeMediaFile(String filename, byte[] data) throws IOException {
-        return mediaAPI.storeMediaFile(filename, data);
+    public String storeMediaFile(AudioFile audioFile) throws IOException {
+        return mediaAPI.storeMediaFile(audioFile.getFilename(), audioFile.getData());
+    }
+
+    /**
+     * Download the requested audio file from the internet and store it on the disk.
+     * @return The path to the audio file.
+     */
+    public String downloadAndStoreAudioFile(DownloadAudioRequest audioRequest) throws IOException {
+        byte[] data = mediaAPI.downloadMediaFile(audioRequest.getUrl());
+        AudioFile audioFile = new AudioFile();
+        audioFile.setFilename(audioRequest.getFilename());
+        audioFile.setData(data);
+
+        String filePath = storeMediaFile(audioFile);
+        return filePath;
```

Comment:
I believe these changes should be completely moved to `MediaAPI`. My understanding of `IntegratedAPI` is that this class is only used when:
- More than one API is being used (e.g. `addNote`)
- The Android `Context` is needed (e.g. `guiBrowse`)